src/main/java/baritone/behavior/PathingBehavior.java

```diff
@@ -372,6 +377,16 @@ public CalculationContext secretInternalGetCalculationContext() {
         return context;
     }
 
+    public Optional<Double> estimatedTicksToGoal() {
+        if (goal == null) {
+            return Optional.empty();
+        }
+        BetterBlockPos currentPos = ctx.playerFeet();
+        double current = goal.heuristic(currentPos.x, currentPos.y, currentPos.z);
+        double start = goal.heuristic(startPosition.x, startPosition.y, startPosition.z);
+        return Optional.of(current * ticksElapsedSoFar / (start - current));
```

Comment:
Can this cause an issue by dividing by zero? If I start a path then immediately do ETA, `start - current` would be zero. What happens then?

Reply:
user error 😂

Reply:
No, then should return `Optional.empty()` because there is not ETA

Reply:
Also this might happen with clients calling this every tick (e.g. for a hud)

Reply:
Should it return `Optional.empty()` whenever the path has just started or only if `start - current == 0`?

Reply:
Can Baritone be following a path while calculating a primary segment? Because while thinking about this I noticed that I expected `estimatedTicksToGoal` to only be called when `startPosition` and `ticksElapsedSoFar` are both set for the same path, but the former is set before pathing and the latter after pathing.